hugolib/site.go

```diff
@@ -105,7 +106,8 @@ type SiteInfo struct {
 	Authors               AuthorList
 	Social                SiteSocial
 	Sections              Taxonomy
-	Pages                 *Pages
+	Pages                 *Pages // Includes only pages in this language
+	TranslatedPages       *Pages // Includes other translated pages, excluding those in this language.
```

Comment:
ok I found it.. `site_test.go` reveals the issue.  Basically, with content files like this (where `DefaultLang = "en"`:

    content/doc1.en.md
    content/doc1.fr.md
    content/doc2.fr.md

The `doc2.fr.md` file will be totally unreachable, as `.Pages` only lists the current language's pages (assuming we're rendering the `en` site).

In that case, we could have `AllPages` as you suggested.. where you'd filter out.. That would do the trick.